cli/src/commands/simplify_parents.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashSet;
+
+use itertools::Itertools;
+use jj_lib::backend::BackendResult;
+use jj_lib::revset::RevsetExpression;
+use jj_lib::rewrite::CommitRewriter;
+use jj_lib::settings::UserSettings;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Simplify parent edges for the specified revision(s).
+///
+/// Removes all redundant parent edges going into any of the specified
+/// revisions. By definition, this has no effect on any commit contents and
+/// should not cause any working copy changes or conflicts.
+///
+/// In other words, for all (A, B, C) where A has (B, C) as parents and C is an
+/// ancestor of B, A will be rewritten to have only B as a parent instead of
+/// B+C.
+#[derive(clap::Args, Clone, Debug)]
+#[command(group = clap::ArgGroup::new("revision-args").multiple(true).required(true))]
+pub(crate) struct SimplifyParentsArgs {
+    /// Simplify specified revision(s) together with their trees of descendants
+    /// (can be repeated)
+    #[arg(long, short, group = "revision-args")]
+    sources: Vec<RevisionArg>,
+    /// Simplify specified revision(s) (can be repeated)
+    #[arg(long, short, group = "revision-args")]
+    revisions: Vec<RevisionArg>,
+}
+
+pub(crate) fn cmd_simplify_parents(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &SimplifyParentsArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let revs = RevsetExpression::descendants(
+        workspace_command
+            .parse_union_revsets(ui, &args.sources)?
```

Comment:
Relatedly to the previous comment, I think this should use `resolve_some_revsets_default_single`.

Reply:
A bit off topic, but do you remember why `rebase -s`/`-b` requires `all:`? I don't think `-s multiple-roots` is a footgun. `-s`/`-b` are similar to `-r`.

Reply:
`-s multiple-roots` can be a footgun, at least if it acts in the way that makes sense to me.

I'd like `-s A -s B -d X` to be equivalent to `-s all:A|B -d X` and result in both `A` and `B` becoming direct children of `X`, *independently* of whether one of `A` or `B` is a descendant of the other one.

If so, `rebase -s A::` would give a very confusing result, and so I'd prefer that gave an error, and the user did `rebase -s all:A::` if that's what they wanted.

There might be other ways to negotiate this problem, which would involve changing what `-s multiple_revision` means (e.g. automatically rewrite it to `-s roots(multiple_revisions)`, making `-s A|B` result in different thing if `B` is a child of `A`). However, the behavior I described seems sensible to me and seems to require `all:`.

Reply:
Though, this makes me think that all of my arguments for `rebase` do not apply to `simplify-parents`. So, maybe we could keep using `parse_union_revset` here. I'm not sure whether or not this slight extra flexibility is worth making the UI subtly different from `rebase`'s.

I think, for `simplify-parents`, I still mildly prefer making the UI the same as `rebase`'s, but I no longer think it's particularly important.

Reply:
> I'd like `-s A -s B -d X` to be equivalent to `-s all:A|B -d X` and result in both `A` and `B` becoming direct children of `X`, _independently_ of whether one of `A` or `B` is a descendant of the other one.

Ah, good point. For `-s A|B`, the user might expect `roots(A|B)` to be the destination children (if `-s` accepted revision ranges by default.)

> Though, this makes me think that all of my arguments for `rebase` do not apply to `simplify-parents`.

Yes. `simplify-parents` is similar to `jj fix -s`.

Reply:
>  For -s A|B, the user might expect roots(A|B) to be the destination children

Yes, or `roots(connected(A|B)`. Not having to choose between these slightly different behaviors is one advantage of the current behavior.